README.md

```diff
@@ -24,39 +30,90 @@ See [Changelog](CHANGELOG.md)
+`target`: The numeric ID of the target node for the edge.
 
-    data-processor/datasets/sample/network.dot
+Creating and editing the network.dot file is recommended to be done by using NetworkX, a popular Python package for creating, manipulating, and analyzing graphs and networks. NetworkX provides a wide range of functionality for working with graphs, including algorithms for graph traversal, centrality, clustering, and more. Additionally, NetworkX provides a `write_dot` function that can be used to export the graph in the Graphviz DOT format, which can be used by the Algorithm. For Javascript libraries such as graphlib can be used to achieve the same to create or export the graph in the Graphviz DOT format. Its importance to the ZMLT algorithm implementation cannot be overstated, as it serves as the backbone for generating visually appealing and informative graph layouts.
 
-This location can be changed by changing the value of $NETWORK environment variable in [config.sh](data-processor/datasets/sample/config.sh)
+for example refer [json2dot.py](https://github.com/cns-iu/obms/blob/main/map4sci/src/json2dot.py). This script takes a JSON file as input and generates a DOT file representing the network described in the JSON file. To use the script, simply run `python json2dot.py data.json network.dot` where data.json is the JSON file containing the network data and network.dot is the graph in the DOT format.
 
-All the `config.sh` files should follow the format as given in [config.example.sh](data-processor/datasets/config.example.sh)
 
-### 5. Run
+### 3. Run
 
 Run [run.sh](data-processor/run.sh) file to run all the steps in the algorithm. This script runs all the scripts in the [scripts](data-processor/scripts) folder one by one.
 
     cd data-processor
     ./run.sh
 
+### Running using Docker
+
+1. Install Docker.
+2. Clone the Map4Sci repository: `git clone https://github.com/cns-iu/map4sci`.
+3. `cd map4sci`.
+4. Build the Docker image: `docker build -t map4sci`.
+5. Start a Docker container from the image: `docker compose run -it map4sci`. This will run the run.sh script automatically.
+
+You can include optional environment variables when starting the container to customize the behavior of the run.sh script. Here are the available options:
+
+`CURRENT_DATASET`: The name of the dataset to use for the workflow. This should match one of the dataset names in the data directory. Default is Sample.
+`CURRENT_VERSION`: The version of the dataset to use for the workflow.
+`NODE_OPTIONS`: Additional Node.js options to pass to the node command.
+
+### Running using Docker Compose
+
+1. Install Docker
+2. Clone the Map4Sci repository:`git clone https://github.com/cns-iu/map4sci`
+3. `cd map4sci`
+4. Run the following command to start the containers: `docker-compose up`
+5. Wait for the containers to start up. This may take a few minutes, depending on your machine
+6. It should now start running for the sample dataset in the repository
```

Comment:
Add section like with docker and cwl that says how to update/customize (ie modify the docker-compose.yml file analagously)